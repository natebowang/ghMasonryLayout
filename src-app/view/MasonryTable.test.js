import React from 'react';
import renderer from 'react-test-renderer';
import Table from './MasonryTable';

const mockRenderItem = (item) => {
    return <div>{item}</div>
};

const mockColumnWidth = 20;
const mockColumnNo = 4;
const mockMatrix = [[0, 4, 8, 12], [1, 5, 9, 13], [2, 6, 10, 14], [3, 7, 11]];
const mockItems = [...Array(15).keys()]; // [0,1, ... ,15]

// snapshot test with react-test-renderer
test("App changes the class when hovered", () => {
    const component = renderer.create(
        <Table renderItem={mockRenderItem}
               matrix={mockMatrix}
               columnWidth={mockColumnWidth}
               columnNo={mockColumnNo}
               items={mockItems}
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});



