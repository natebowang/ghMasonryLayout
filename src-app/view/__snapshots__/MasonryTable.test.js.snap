// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App changes the class when hovered 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-evenly",
      "padding": "0.6rem",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "alignSelf": "flex-start",
        "display": "flex",
        "flexDirection": "column",
        "width": "20rem",
      }
    }
  >
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        0
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        4
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        8
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        12
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "alignSelf": "flex-start",
        "display": "flex",
        "flexDirection": "column",
        "width": "20rem",
      }
    }
  >
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        1
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        5
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        9
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        13
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "alignSelf": "flex-start",
        "display": "flex",
        "flexDirection": "column",
        "width": "20rem",
      }
    }
  >
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        2
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        6
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        10
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        14
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "alignSelf": "flex-start",
        "display": "flex",
        "flexDirection": "column",
        "width": "20rem",
      }
    }
  >
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        3
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        7
      </div>
    </div>
    <div
      style={
        Object {
          "padding": "0.6rem",
        }
      }
    >
      <div>
        11
      </div>
    </div>
  </div>
</div>
`;
